app/controllers/profiles_controller.rb

```diff
@@ -41,28 +33,48 @@ def destroy
 
   private
 
-  def profile
-    @profile ||= Profile.new
+  def find_profile
+    @profile = Profile.find params[:id]
   end
 
-  def find_profile
-    @profile = Profile.find(params[:id])
+  def find_profile_user
+    find_profile.try(:user) || User.new
   end
 
   def new_profile_form
-    @profile_form ||= ProfileForm.new profile
+    @profile_form ||= ProfileForm.new profile: Profile.new, user: User.new
+  end
+
+  def edit_profile_form
+    @profile_form ||= ProfileForm.new profile: find_profile, user: find_profile_user, has_associated_user: has_associated_user
   end
 
+  def has_associated_user
+    find_profile.try(:user).present?
+  end
+
+  def create_and_save
+    new_profile_form.validate(profile_params) && new_profile_form.save
```

Comment:
Not sure about all these `foo_and_bar` actions. These call two methods on the form object, it feels like the form object should just expose an API capable of handling this.